AraEvent/AraEventCalibrator.cxx

```diff
@@ -896,226 +1030,373 @@ void AraEventCalibrator::calibrateEvent(UsefulAtriStationEvent *theEvent, AraCal
+   
+            //! Set the iterators to point to the correct channel in the map
             timeMapIt=theEvent->fTimes.find(chanId);
             voltMapIt=theEvent->fVolts.find(chanId);
 
             int samp=0;
-            int index=0;
-            // Now loop over the 64 samples
-            Double_t tempTimes[SAMPLES_PER_BLOCK];
-            Double_t tempVolts[SAMPLES_PER_BLOCK];
-            Int_t voltIndex[SAMPLES_PER_BLOCK];
-
-        
-            // Here is the Epsilon calibration
-            if(AraCalType::hasBinWidthCalib(calType)) {
-                if(!firstTime) {
-                    // Add on the time between blocks
-                    // time+=fAtriEpsilonTimes[dda][chan][capArray];    //I think this is not needed anymore. The actual block timing is given in the calibration file. -THM-
-                    // if(dda==1 && chan==1)
-                    // std::cerr << "Block " << time << "\t" << fAtriEpsilonTimes[dda][chan][capArray] << "\n";
-                }
-            }
+            //! Now loop over the 64 samples
             for(shortIt=vecVecIt->begin();
                 shortIt!=vecVecIt->end();
                 shortIt++){
 
-                if(AraCalType::hasBinWidthCalib(calType)) {
+                time+=NSPERSAMP_ATRI;
+                timeMapIt->second.push_back(time); ///< Filling with time
+                voltMapIt->second.push_back((*shortIt)); ///< Filling with volts
+                sampleList->at(chanId).push_back(block * samples_per_block + samp); ///< Filling with sample number. It is needed for pedestal subtraction and voltage calibration
+                samp++;
 
-                    // Now need to work out where to place the sample
-                    voltIndex[samp]=fAtriSampleIndex[dda][chan][capArray][samp];
+            }
+        }
+    }
+}
 
-                    // Now get the time
-                    // tempTimes[samp]=time+fAtriSampleTimes[dda][chan][capArray][samp];   //replaced with line below -THM-
-                    //    A new version of timing calibration: every block starts nominally at multiples of 20. The single samples though are referenced to the TSA-strobe
-                    // which happens every 40.0 ns. The odd blocks (capArray=1) T need to have 20ns subtracted then. The even blocks (capArray=0) wont be modified  -THM-
-                    //RJN -- Now confusingly the [samp] in tempTimes actually refers to the valid sample time
-                    // (i.e if there are 33 valid samples only the first 33 entries in this array will be valid)
-                    tempTimes[samp]=blockInEvent[dda]*20.0 + fAtriSampleTimes[dda][chan][capArray][samp] - 20.0*capArray;
+/*! 
+    Timing calibration
+    This step calibrates the time of each sample and only selecting the samples that have good performance
+    fAtriSampleIndex and fAtriSampleTimes stores the sample numbers and times that have good performance
+    WF will be trimmed by these tables
```

Comment:
What does "trimmed" mean?

Reply:
The changes are in the code.... some how it is not showing in here??????

Reply:
Could you check the line 1116 in the filed changed menu? I can see from there